rfcs/document_all_media_properties/20230307-project_proposal.md

```diff
@@ -0,0 +1,147 @@
+Project proposal for
+[Document all media properties](https://github.com/WordPress/openverse/issues/412)
+
+## Due date:
+
+2023-03-21
+
+## Assigned reviewers
+
+- [ ] @AetherUnbound
+- [ ] @krysal
+
+## Description
+
+### Year Goal
+
+Overcoming **Data Inertia**.
+
+### Motivation
+
+The first step in making the data we have reliable is describing it in detail.
+The main goal of this project is to establish a comprehensive baseline of our
+media data: what information we currently have about each media item and what we
+expect to have. Our database currently holds more than 700 million rows of media
+data. The same data is in our API database and Elasticsearch, but the data
+properties there are slightly different. Due to the project's history, some data
+in the catalog does not meet our current requirements and lacks necessary
+fields, while other fields store data that doesn't align with user
+expectations[^1].
+
+By documenting all data properties, we can ensure that we have a complete
+understanding of the data. This documentation will improve the API user and the
+Openverse maintainer experience. It will also be the basis for data
+normalization which will lead to faster searches and a better experience for
+catalog and API contributors, as well as API users.
+
+### Success criteria
+
+The project will be considered successful when the document described below is
+created, reviewed by at least two maintainers, and published on the Openverse
+documentation site.
+
+### Implementors
+
+@rwidom - creator of the
+[image table](https://docs.google.com/spreadsheets/d/1gaVsvFnsYby2iwzRm0Ta9IPhvNixCpQdHByGLo1r_pg/edit?usp=sharing)
+("image table ddl" tab)
+
+@obulat - main implementor @openverse-team - reviewers
+
+### Implementation details
+
+As this is mainly a documentation project, its implementation plan can be
+written here, inside the project plan. (I think splitting it will unnecessarily
+slow down a smallish project. Please, correct me if you think I'm wrong) This
+project will create a **table** with all the media properties that we store or
+compute on all levels of the stack (Catalog, API database, API elasticsearch,
+Frontend).
+
+#### Table as a living document
+
+This table will serve as a basis for any work on data normalization work. As
+such, it will be a work in progress because the links to the code and the
+details about the properties should change as we progress through the data
+normalization project. To ensure that the table is up-to-date, this project will
+also add automation to the catalog CI pipeline to detect changes to the fields
+in the table and the catalog DDL and add a comment to update them, similar to
+the
+[migration safety warning](https://github.com/WordPress/openverse/blob/main/.github/workflows/migration_safety_warning.yml)
```

Comment:
Agreed, while this would be ideal to run GitHub for access to the repo code, it would not be ideal if we needed to access prod resources as part of it. Since we already have all the connection processes set up for Airflow, I think a DAG sounds great for this purpose!